.pre-commit-config.yaml

```diff
@@ -0,0 +1,14 @@
+repos:
+-   repo: https://github.com/pycqa/isort
+    rev: 5.10.1
+    hooks:
+    - id: isort
+      name: isort (python)
+-   repo: https://github.com/ambv/black
+    rev: 22.3.0
+    hooks:
+    - id: black
+-   repo: https://gitlab.com/pycqa/flake8
+    rev: 4.0.1
+    hooks:
+    - id: flake8
```

Comment:
Good idea to set up pre-commit.
It covers most of the good repos, some more that I think would be useful. (Pasting the .pre-commit-config.yaml I usually use)
```yaml
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.2.0
    hooks:
      - id: end-of-file-fixer
      - id: trailing-whitespace
      - id: debug-statements
      - id: check-merge-conflict

  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: v2.6.2
    hooks:
      - id: prettier

  - repo: https://github.com/python/black
    rev: 22.3.0
    hooks:
      - id: black
        language_version: python3

  - repo: https://github.com/pycqa/isort
    rev: 5.10.1
    hooks:
      - id: isort
        args: ["--settings", "setup.cfg"]

  - repo: https://gitlab.com/pycqa/flake8
    rev: 3.9.2
    hooks:
      - id: flake8
        args: ["--config", "setup.cfg"]
```
(The versions may be off, pick the latest ones)

The setup.cfg is this
```
[isort]
profile=black

[flake8]
extend-ignore = E203 #https://github.com/psf/black/issues/315#issuecomment-395457972
max-line-length = 128
per-file-ignores = __init__.py: F401
```
These are useful for [`isort` to be compatible with `black`](https://pycqa.github.io/isort/docs/configuration/black_compatibility.html), and the ones for `flake8` are useful in general once the project grows.

Feel free to pick and chose what you find relevant.
cc: @veds12